src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoSearchIT.java

```diff
@@ -508,4 +509,321 @@ public void shouldNotFilterByGatewayAccountIdWhenNotSpecified() {
+        transactionFixture = aTransactionFixture()
+                .withLastDigitsCardNumber("4242")
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(1));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setLastDigitsCardNumber("4242");
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
+
+        assertTransactionEquals(transactionList.get(0), transactionFixtureChild);
+        assertTransactionEquals(transactionList.get(0).getParentTransactionEntity(), transactionFixture);
+        assertTransactionEquals(transactionList.get(1), transactionFixture);
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(2L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByCardBrandOnTransactionAndParentTransaction() {
+        transactionFixture = aTransactionFixture()
+                .withCardBrand("visa")
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(2));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setCardBrands(new CommaDelimitedSetParameter("visa"));
+
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
+        assertTransactionEquals(transactionList.get(0), transactionFixtureChild);
+        assertTransactionEquals(transactionList.get(1), transactionFixture);
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(2L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByFromDate() {
+        transactionFixture = aTransactionFixture()
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(1));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setFromDate(now(ZoneOffset.UTC).toString());
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
+
+        assertTransactionEquals(transactionList.get(0), transactionFixtureChild);
+        assertTransactionEquals(transactionList.get(0).getParentTransactionEntity(), transactionFixture);
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(1L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByToDate() {
```

Comment:
[minor] Possibly the name should include that it will not include parent transaction, because it's out of search range.